src/bvh/bvh.rs

```diff
@@ -22,34 +20,55 @@ use ray::Ray;
 /// [`AABB`]: ../aabb/struct.AABB.html
 /// [`BVH`]: struct.BVH.html
 ///
+#[derive(Debug, Copy, Clone)]
 pub enum BVHNode {
     /// Leaf node.
     Leaf {
-        /// The shapes contained in this leaf.
-        shapes: Vec<usize>,
+        /// The node's parent.
+        parent: usize,
+
+        /// The node's depth
+        depth: u32,
+
+        /// The shape contained in this leaf.
+        shape: usize,
     },
     /// Inner node.
     Node {
+        /// The node's parent.
+        parent: usize,
+
+        /// The node's depth
+        depth: u32,
+
         /// The convex hull of the shapes' `AABB`s in child_l.
         child_l_aabb: AABB,
 
         /// Left subtree.
```

Comment:
The comments are a little outdated, this is not the subtree itself, it's the index of the subtree.